File: descend_calc/descend_calc.rb
def get_input(msg)
    loop do
        print msg
        input = gets.chomp
        if input =~ /^-?\d+$/
            return input.to_i
        end
    end
end

def main
    current_altitude = get_input("Current altitude: ")
    airport_altitude = get_input("Airport altitude: ")

    altitude_to_lose = ((current_altitude - airport_altitude).to_f / 100.0) / 3.0

    current_speed = get_input("Current speed: ")
    avg_dsc_speed = get_input("Average descend speed: ")

    speed_knots = (current_speed - avg_dsc_speed).to_f / 10.0

    wind_speed = get_input("Wind speed: ").to_f / 10.0
    tail_wind = get_input("Tail or head wind [1 for tail | 0 for head]: ")

    descend_distance = altitude_to_lose + speed_knots + (tail_wind == 1 ? wind_speed : -wind_speed)

    puts "------------------------------"
    puts "Altitude to lose nm: #{altitude_to_lose}"
    puts "Speed nm: #{speed_knots}"
    puts "Wind  nm: #{wind_speed}"
    puts "Total distance: #{descend_distance}"
end

main

File: descend_calc/descend_calc.pl
#!/usr/bin/perl
use strict;
use warnings;

sub get_input
{
    my ($msg) = @_;
    while (1)
    {
        print $msg;
        my $input = <STDIN>;
        chomp $input;
        if ($input =~ /^-?\d+$/)
        {
            return int($input);
        }
    }
}

sub main
{
    my $currentAltitude = get_input("Current altitude: ");
    my $airportAltitude = get_input("Airport altitude: ");

    my $altitudeToLose = (($currentAltitude - $airportAltitude) / 100) / 3;

    my $currentSpeed = get_input("Current speed: ");
    my $avgDscSpeed = get_input("Average descend speed: ");

    my $speedKnots = ($currentSpeed - $avgDscSpeed) / 10;

    my $windSpeed = get_input("Wind speed: ") / 10;
    my $tailWind = get_input("Tail or head wind [1 for tail | 0 for head]: ");

    my $descendDistance = $altitudeToLose + $speedKnots + ($tailWind == 1 ? $windSpeed : -$windSpeed);

    print "------------------------------\n";
    printf "Altitude to lose nm: %.6f\n", $altitudeToLose;
    printf "Speed nm: %.6f\n", $speedKnots;
    printf "Wind  nm: %.6f\n", $windSpeed;
    printf "Total distance: %.6f\n", $descendDistance;
}

main();